.github/workflows/release.yml

```diff
@@ -39,6 +39,16 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/github-token@current
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+      - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+        with:
+          username: ${{ env.GIT_USER }}
+          email: ${{ env.GIT_EMAIL }}
+          token: ${{ env.GITHUB_TOKEN }}
```

Comment:
As far as I can see, the added token has no effect on the workflow.

I guess you are trying to add a GitHub PAT. But your adding it in the workflow.
However, the this job delegates it to a Buildkite job.

Reply:
This is the corresponding Buildkite pipeline https://github.com/elastic/apm-agent-android/blob/main/.buildkite/release.yml

Reply:
The `GITHUB_TOKEN` is the one set up by the `github-token` GitHub Actions.

Reply:
Got it, thanks @reakaleek  - I'll move this to the `.buildkite/release.yml` file then.

@amannocci  I'm not sure I'm following your comment, does that mean that the `setup-git` GitHub action can read the `GITHUB_TOKEN` without me having to pass it in line 51? Or that I'm not passing the token to the `setup-git` action correctly?

Reply:
Sorry, it was the explanation for the following comment:

> As far as I can see the added token has no effect on the workflow.
> 
> I guess you are trying to add a GitHub PAT. But your adding it in the GH workflow. However, this job delegates it to a Buildkite job.

In fact, we are using the `GITHUB_TOKEN` set up by the GitHub Action `github-token` at **L51**.
On another hand, if we use `${{ secrets.GITHUB_TOKEN }}` instead of `${{ env.GITHUB_TOKEN }}` then we will use the auto-generated Github Token set up by GitHub Action.

Reply:
I see thank you. And for my use case (which is to let the buildkite job to push changes straight into the `main` branch as `apmmachine`) is it preferable to use the token from the `github-token` action? Or is it better to use the auto-generated by Github one?

Reply:
the generated token is not able to push to protected branches. See [discussion](https://github.com/orgs/community/discussions/25305).

Hence, you actually have to use the PAT of `apmmachine`

Reply:
Got it, thank you @reakaleek - I've just made the change to move this over to `.buildkite/release.yml`.